colossalai/gemini/memory_tracer/memstats_collector.py

```diff
@@ -150,3 +158,61 @@ def sample_model_data(self) -> None:
     @property
     def cuda_margin_mem(self) -> float:
         return colo_device_memory_capacity(get_current_device()) - max(self.overall_mem_stats('cuda'))
+
+
+class MemStatsCollectorStatic(MemStatsCollector):
+    """
+    A Static Memory statistic collector.
+    """
+
+    def __init__(self, module: nn.Module) -> None:
+        super().__init__()
+        self.module = module
+
+    def init_mem_stats(self, *inputs):
+        self.module.cpu()
+        self.module.train()
+        # data = MetaTensor(torch.rand(int(TM_BATCH_SIZE), 3, 224, 224, device='meta'), fake_device='cuda:0')
+        data = [MetaTensor(torch.rand(inp.shape, device='meta'), fake_device='cpu') for inp in inputs]
+        gm = symbolic_trace(self.module)
+        interp = MetaInfoProp(gm)
+        interp.propagate(data)
+        for node in gm.graph.nodes:
+            self._non_model_data_cuda_list.append(node.meta["fwd_mem_tmp"] + node.meta['fwd_mem_out'])
```

Comment:
Can use the `calculate_fwd_tmp` and `calculate_fwd_out` function in `colossalai.fx.profiler.shard_utils` for the time being

Reply:
Since `MetaInfoProp` suffer from compatibility issues for torch 1.12 and torch 1.11
In torch 1.12 we can keep track of all the tensors as well as their shape.
In torch 1.11 we can only roughly estimate the memory consumption of this layer